mlops/parallelm/mlops/mlops_exception.py

```diff
@@ -1,12 +1,69 @@
+import functools
+import logging
+
 class MLOpsException(Exception):
     """
     Exception that will be returned by the MLOps class on error
     """
     pass
 
+class MLOpsExceptionHandler(object):
+    handlers = {}
+    exceptions_list = []
+    exceptions = ()
+    _logger = logging.getLogger(__name__)
 
-class MLOpsException(Exception):
-    """
-    Exception to be returned by mlops api functions on error
-    """
-    pass
+    @classmethod
+    def _warn(cls, e):
+        cls._logger.warning("suppressed exception: {}".format(str(e)))
+
+    @staticmethod
+    def _rethrow(e):
+        raise Exception("original exception: {} {}".format(type(e), str(e)))
+
+    def __init__(self):
+        pass
+
+    def __call__(self, func):
```

Comment:
Answering also for: AFAIK, no need for the brackets "()".

This code defines decorator with arguments, even though I don't use any arguments. If I define decorator with arguments, it is used with brackets: @MLOpsExceptionHandler()

I had to do it this way for decorating class methods, otherwise i'm not receiving `self` in the decorated method.

Appreciate if you could help me to figure out what is going on.

https://realpython.com/primer-on-python-decorators/
https://wiki.python.org/moin/PythonDecoratorLibrary, Different Decorator Forms